### ISSUE TITLE
Support for different platforms

### ISSUE THREAD

**Issue by Akendo** [guedou/cargo-strip#1]
I'm trying to integrate this strip tools into my build chain. However, it seems to support only x86 in the moment. My MIPS platform remains 'unstriped'. 
```
 $ cargo build --target=mips-unknown-linux-musl
 $ cargo strip
```
Nothing changed. With the default platform:
```
 $ cargo build --release
 $ cargo strip
"/tmp/NetStats/target/release/NetStats" stripped (reduced by 699 kB)!
```

best regards,
akendo

**Comment by guedou** [comment]
Thanks for your interest in `cargo-strip` !

Currently, the tool uses the output of  `cargo metadata` to identify the binaries that needs to be striped and do not search them in `./release/mips-unknown-linux-musl`.

I will try to implement this feature with a command line argument such as:
```
cargo strip --target=mips-unknown-linux-musl
```

**Comment by Akendo** [comment]
That's great. I think one way would be implementing different types for the `strip` binary. I guess the users need to provided in one way the path to the `strip`. Not sure right know who this should be done.

**Comment by guedou** [comment]
Any `strip` should be able to remove the symbols from an ELF binary. I am not aware of architecture dependant `strip`.

**Comment by mtpiercey** [comment]
Just thought I'd pop in here with a random example of what I think @Akendo means.

I've been writing Rust code for my Raspberry Pi (Zero W, ARM6), and in order to strip the binaries I make with cargo, I need to do something like:

`arm-linux-gnueabihf-strip target/arm-unknown-linux-gnueabihf/debug/program-name`

I learned that from this blog post: https://dev.to/jeikabu/raspberry-pi-zero-raspbian-rust-primer-3aj6

---
Turns out, running `strip` on the binaries for my Pi comes up with:
```sh
strip: Unable to recognise the format of the input file `target/arm-unknown-linux-gnueabihf/debug/program-name'
```

I too would be interested in `cargo-strip` supporting different build targets (though admittedly I'm not fluent enough in Rust to do really anything to help).

**Comment by guedou** [comment]
Could you try PR #4? It adds the `--target` argument to `cargo-strip`, and according to the unit tests it works as expected.

**Comment by mtpiercey** [comment]
Looks good! Using it with my `arm-unknown-linux-gnueabihf` binary, it got `(reduced by 3830 kB)!`

**Comment by guedou** [comment]
Awesome, I will release 0.2.2 shortly.

**Comment by guedou** [comment]
I merged PR #2.

**Comment by Akendo** [comment]
Thanks a lot, I'm going to test it right away!
